static/js/components/EditFollowupRequestDialog.jsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import Dialog from '@material-ui/core/Dialog';
+import DialogContent from '@material-ui/core/DialogContent';
+
+import FollowupRequestForm from './FollowupRequestForm';
+
+
+const EditFollowupRequestDialog = ({ followupRequest, instrumentList, instrumentObsParams }) => {
+  const [open, setOpen] = useState(false);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  return (
+    <span key={followupRequest.id}>
+      <button type="button" onClick={handleClickOpen}>
+        Edit
+      </button>
+      <Dialog open={open} onClose={handleClose} style={{ position: "fixed" }}>
+        <DialogContent>
+          <FollowupRequestForm
+            source_id={followupRequest.source_id}
+            action="editExisting"
+            followupRequest={followupRequest}
```

Comment:
You should be able to unpack these automatically without specifying each field.

Reply:
Can you please explain? This is just standard props-passing

Reply:
```
<ChildComponent {...props} />
```

Reply:
Only 3 of the 7 props being passed into `<ChildComponent>` are also props of `<ParentComponent>` (`followupRequest, instrumentList, instrumentObsParams`), and since we follow the object destructuring pattern for props in our functional component definitions, this isn't available here.

Reply:
Sure, I changed the function signature to use a single props object and not use destructuring so that we can do `{...props}` here.

Reply:
Actually, that's an anti-pattern - reverting to what I had done originally which is the correct form. 
```
/home/travis/build/skyportal/skyportal/static/js/components/EditFollowupRequestDialog.jsx
  21:16  error  Must use destructuring props assignment  react/destructuring-assignment
  28:24  error  Must use destructuring props assignment  react/destructuring-assignment
  32:13  error  Prop spreading is forbidden              react/jsx-props-no-spreading
```

Reply:
Ooh forbidden! :fire: